src/Expression/ViolationDescription.php

```diff
@@ -0,0 +1,38 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Arkitect\Expression;
+
+class ViolationDescription
+{
+    /** @var string */
+    private $rule;
+    /** @var string|null */
+    private $violation;
+
+    private function __construct(string $rule, ?string $violation)
+    {
+        $this->rule = $rule;
+        $this->violation = $violation;
+    }
+
+    public static function withDescription(ExpressionDescription $brokenRule, string $description): self
```

Comment:
I think that we can improve the name of this class without using two times `Description`: in the class name and in the method name. What do you think?

Reply:
Agreed. How about ViolationMessage?

Reply:
Sounds good to me